### DISCUSSION TITLE
frontmatter.head 中添加的 link.icon 怎么读取当前目录下的 favicon.ico 文件？

### DISCUSSION THREAD

**Discussion by big-dream** [vuepress/core#1567] [Q&A] [answered]
我想在 `index.md` 利用 [frontmatter.head](https://v2.vuepress.vuejs.org/zh/reference/frontmatter.html#head) 为当前 md 文档设置一个 favicon，路径为当前目录下的 `favicon.ico` 文件。

目录结构
```
- docs
-- index.md
-- favicon.ico  
```

index.md 文件内容：
```markdown
---
head:
  - [link, { rel: 'icon', href: 'favicon.ico' }]
---
# title

![icon](favicon.ico)

contents...
```

文档 `build` 之后，生成的 `link` 标签为 `<link rel="icon" href="240425.jpg">`，`favicon.ico` 对应的路径应该是 `/assets/favicon-_QuWoUl2.ico`。

请问我应该怎么写才可以让`link`标签的`href`正确的解析为当前目录下的 `favicon.ico` 文件？

**Comment by Mister-Hope** [comment, marked as answer]
放在public，以保持原路径

**Comment by big-dream** [comment]
放 public 就跟其它图片分开管理了，不知道能不能跟 `![icon](favicon.ico)` 达到一样的效果。如果实在是不能，那就只能考虑放在 public 了。

**Reply by Mister-Hope** [reply]
任何在源文件的文件如果需要生效都需要经过打包器处理，其输出地址默认也会按规则输出。

换言之，本质上你在frontmatter中设置的内容，打包器也不会知道它是一个需要处理的资源文件。因此这种思路本质上就是错误的。

**Reply by big-dream** [reply]
好的，谢谢大佬
